File: client/src/types/Sprint.ts
export type SprintStatus = "Planned" | "Active" | "Completed";

export interface Sprint {
  id: number;
  projectId: number;
  name: string;
  goal: string | null;
  startDate: string;
  endDate: string;
  status: SprintStatus;
  issueCount: number;
}

export interface CreateSprintPayload {
  name: string;
  goal: string | null;
  startDate: string;
  endDate: string;
}

export interface UpdateSprintPayload {
  name: string;
  goal: string | null;
  startDate: string;
  endDate: string;
}

File: server/SprintBoard.Api/Models/Sprint.cs
namespace SprintBoard.Api.Models;

public class Sprint
{
    public int Id { get; set; }

    public int ProjectId { get; set; }
    public Project Project { get; set; } = null!;

    public string Name { get; set; } = string.Empty;

    public string? Goal { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime EndDate { get; set; }

    public SprintStatus Status { get; set; } = SprintStatus.Planned;

    public List<Issue> Issues { get; set; } = new();
}